### PR TITLE
CBL-431: ContainsUTF8 for Apple

### PR DESCRIPTION
**Pull request by jayahariv**
* trying to breakdown this task to three. Apple, ICU, and win. 
* this PR include Apple implementation for ContainsUTF8 check.

### CODE REVIEW

File: LiteCore/tests/SQLiteFunctionsTest.cc [hunk 1]
```diff
@@ -530,3 +530,76 @@ N_WAY_TEST_CASE_METHOD(SQLiteFunctionsTest, "SQLite collation", "[Query][Collati
...
+        {"abÜcd"_sl, "Ü"_sl,         0, true, true },
+
+        // Case sensitive, diacritic insensitive:
+        {"Aabcd"_sl, "ABC"_sl,      -1, true, false},
+        {"test á"_sl, "test a"_sl,   0, true, false},
+        {"testá"_sl, "test á"_sl,   -1, true, false},
+        {"testábcd "_sl, "á"_sl,     0, true, false},
+        {"test á"_sl, "test A"_sl,  -1, true, false},
+        {"test á"_sl, "test Á"_sl,  -1, true, false},
+        {"u"_sl, "Ü"_sl,             0, true, false},
```
Comment (jayahariv) [inline LiteCore/tests/SQLiteFunctionsTest.cc]:
```suggestion
        {"U"_sl, "Ü"_sl,             0, true, false},
```

**Comment by jayahariv** [comment]
windows test is failing due to other reason, not related with this PR.
```
....
Receiving objects:  13% (3967/30510)
fatal: write error: No space left on device
```

**Comment by borrrden** [comment]
I don't want to merge this for Mercury if the others are not going to be ready, just FYI.

**Comment by jayahariv** [comment]
Closing this PR, will reopen once ready with other implementation. We already have reference to this PR in the Jira ticket, so we can re-use the code next time.

**Pull request closed without merging**